src/system/Form/RadioBox.js

```diff
@@ -0,0 +1,66 @@
+/** @jsxImportSource theme-ui */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as ReactToggleGroup from '@radix-ui/react-toggle-group';
+import { Heading, Text } from '@automattic/vip-design-system';
+import { MdCheckCircle } from "react-icons/md";
+
+/**
+ * Internal dependencies
+ */
+
+const RadioBox = ( { onChange, value, options, ...props } ) => (
+	<ReactToggleGroup.Root
+		onValueChange={ onChange }
+		value={ value }
+		type="single"
+		{ ...props }
+	>
+		{
+			options.map( option => (
+				<ReactToggleGroup.Item
+					key={ option.value }
+					value={ option.value }
+					sx={ {
+						p: 3,
+						background: 'none',
+						cursor: 'pointer',
+						borderRadius: 2,
+						textAlign: 'left',
+						border: '1px solid',
+						borderColor: 'border',
+						position: 'relative',
+						svg: {
+							display: 'none'
+						},
+						'&:hover': {
+							borderColor: 'grey.10',
+						},
+						'&[data-state=on]': {
+							svg: {
+								display: 'block'
+							},
+							borderColor: 'primary',
+						},
+					} }
+				>
+					<MdCheckCircle size={ 16 } sx={ { position: 'absolute', top: 2, right: 2, color: 'primary' } } />
+					<Heading variant="h4" sx={ { mb: 0 } }>{ option.label }</Heading>
```

Comment:
Channeling Gary here: using headings here is somewhat non-semantic and we should probably use proper `<label>` tags instead?